econml/dml.py

```diff
@@ -130,18 +146,31 @@ def const_marginal_effect(self, X=None):
             X = np.ones((1, 1))
         return self._model_final.predict(X)
 
-    # need to override effect in case of discrete treatments
+    def const_marginal_effect_interval(self, X=None, *, alpha=0.1):
+        if X is None:
```

Comment:
Why isn't this logic of checking if X is None and replacing it implemented at a higher class level? Maybe we can have some sort of _check_input_arrays() utility function that implemtns this (similar to sklearn). This could also potentially also handle the discrete treatment transformations.

Reply:
I agree that moving more logic up the hierarchy would be valuable (though perhaps controlled by flags in case it's not appropriate).  However, I think that might be better taken care of in a separate PR.